src/Raven.Server/Config/Categories/IndexingConfiguration.cs

```diff
@@ -404,6 +404,12 @@ public virtual bool RunInMemory
         [IndexUpdateType(IndexUpdateType.Reset)]
         [ConfigurationEntry(Constants.Configuration.Indexes.IndexingStaticSearchEngineType, ConfigurationEntryScope.ServerWideOrPerDatabaseOrPerIndex)]
         public SearchEngineType StaticIndexingEngineType { get; protected set; }
+
+        [Description("Corax index compression max documents used for dictionary creation.")]
+        [DefaultValue(100000)]
```

Comment:
You said that `10_000` is enough, no?

Reply:
Maybe the actual number of items (docs/counters/time series) that we'll process for the dictionary creation should be adjusted to the size of a collection that an index will work on instead of hard coded value?

The issue is that for collections <100K we'll process everything twice - once during dict creation and second time during the actual indexing batch.

Another thing we should take into account is encryption - then we have limited size of indexing batches while during the dictionary creation we don't take that into account. This means that we'll process 100K items at once and will have very high unmanaged memory usage leading to OOM errors, especially if multiple indexes will be deployed at the same time.

Reply:
We can "forget" the document after load, no? That should handle this.